shared-bin/src/network.rs

```diff
@@ -0,0 +1,134 @@
+use std::path::PathBuf;
+
+use clap::Parser;
+use clap::Subcommand;
+use daemon::bdk::bitcoin;
+use daemon::bdk::bitcoin::Address;
+use daemon::bdk::bitcoin::Amount;
+
+const MAINNET_ELECTRUM: &str = "ssl://blockstream.info:700";
+const TESTNET_ELECTRUM: &str = "ssl://blockstream.info:993";
+
+#[derive(Parser, Clone)]
+pub enum Network {
+    /// Run on mainnet (default)
+    Mainnet {
+        /// URL to the electrum backend to use for the wallet.
+        #[clap(long, default_value = MAINNET_ELECTRUM)]
+        electrum: String,
+
+        #[clap(subcommand)]
+        withdraw: Option<Withdraw>,
+    },
+    /// Run on testnet
+    Testnet {
+        /// URL to the electrum backend to use for the wallet.
+        #[clap(long, default_value = TESTNET_ELECTRUM)]
+        electrum: String,
+
+        #[clap(subcommand)]
+        withdraw: Option<Withdraw>,
+    },
+    /// Run on signet
+    Signet {
+        /// URL to the electrum backend to use for the wallet.
+        #[clap(long)]
+        electrum: String,
+
+        #[clap(subcommand)]
+        withdraw: Option<Withdraw>,
+    },
+    /// Run on regtest
+    Regtest {
+        /// URL to the electrum backend to use for the wallet.
+        #[clap(long)]
+        electrum: String,
+
+        #[clap(subcommand)]
+        withdraw: Option<Withdraw>,
+    },
+}
+
+impl Default for Network {
+    fn default() -> Self {
+        Network::Mainnet {
+            electrum: MAINNET_ELECTRUM.to_string(),
+            withdraw: None,
+        }
+    }
+}
+
+#[derive(Subcommand, Clone)]
+pub enum Withdraw {
```

Comment:
🔧 I think the withdraw subcommand doesn't belong in the `network` module. Perhaps we should create a `cli` module and put it there.

Reply:
just to be clear: would renaming `network` to `cli` be acceptable?

Reply:
I think `Network` and `Withdraw` don't belong together.

Reply:
well, `Withdraw` is *part* of `Network` right now, isn't it? 🤔

I will move `Withdraw` to a separate `withdraw` module for now as a first step towards separating them.

Reply:
Nvm, your proposal is better: renaming `network` to `cli`. I failed to notice that `Network` was actually a CLI subcommand. Sorry for the confusion!